lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0..2].join(' | ')} "
+    puts "------------"
+    puts " #{@cells[3..5].join(' | ')} "
+    puts "------------"
+    puts " #{@cells[6..8].join(' | ')} "
+    puts "\n"
+  end
+
+  def first_display
+    puts " 1 | 2 | 3 "
+    puts "------------"
+    puts " 4 | 5 | 6 "
+    puts "------------"
+    puts " 7 | 8 | 9 "
+    puts "\n"
+  end
+
+  def full?
+    @cells.include?(" ") == false
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def taken?(pos)
+    self.position(pos) != " " ? true : false
+  end
+
+  def turn_count
+    @cells.select {|x| x == "X" || x == "O"}.length
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```